### PR TITLE
Fix checks for duplicate metrics (fixes #283)

### LINKED ISSUE #283: Incorrect "duplicate metrics identifier" checks today (2021-03-18)
We just got a whole bunch of emails notifying us:

> Glean: Duplicate metric identifiers detected

Pretty much for any Glean-containing app I think.

Content:

```
Glean has detected duplicated metric identifiers coming from the product 'burnham'.

- 'glean.baseline.duration' defined more than once in glean-android, glean-core
- 'glean.baseline.locale' defined more than once in glean-android, glean-core
- 'os' defined more than once in glean-android, glean-core
- 'os_version' defined more than once in glean-android, glean-core
- 'device_manufacturer' defined more than once in glean-android, glean-core
- 'device_model' defined more than once in glean-android, glean-core
- 'architecture' defined more than once in glean-android, glean-core
- 'client_id' defined more than once in glean-android, glean-core
- 'app_build' defined more than once in glean-android, glean-core
- 'app_display_version' defined more than once in glean-android, glean-core
- 'app_channel' defined more than once in glean-android, glean-core
- 'first_run_date' defined more than once in glean-android, glean-core
- 'glean.error.invalid_value' defined more than once in glean-android, glean-core
- 'glean.error.invalid_label' defined more than once in glean-android, glean-core
- 'glean.error.preinit_tasks_timeout' defined more than once in glean-android, glean-core
```

burnham is a Python tool. It should not depend on `glean-android` at all.
Maybe #278 broke something? (That landed half a day ago, which is why I suspect it)

### PR DESCRIPTION
**Pull request by wlach**

### CODE REVIEW

**Comment by wlach** [comment]
Note: There is an existing check for duplicate checking in:

https://github.com/mozilla/probe-scraper/blob/514e6c37a310ec5a202b6b806cb3d53b466d6918/tests/test_git_scraper.py#L294

... but that's more of an end-to-end test. I think it's valuable to have a more minimal test case.

**Review by fbertsch — APPROVED** [review]
LGTM. The unit tests were a good addition, separate from the integration tests.

File: tests/test_glean_checks.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        ),
+        Repository(
+            "fake-app",
+            dict(BASE_METADATA, dependencies=["glean-core", "glean-android"]),
+        ),
+    ]
+
+
+def test_check_duplicate_metrics_no_duplicates(fake_repositories):
+    # no overlap between metrics defined by glean-core and glean-android (both used by burnham)
```
Comment (badboy) [inline tests/test_glean_checks.py:28]: Oooh, so this was the issue? burnham and the others depends on `glean-deprecated`, which contains both the core metrics and the android ones?

Comment (wlach) [reply tests/test_glean_checks.py:28]: This comment is a little forward looking 😄 . #285 removes glean-deprecated and just adds glean-android explicitly to burnham and friends for backwards compat

Comment (badboy) [reply tests/test_glean_checks.py:28]: Ah, got it. But still, the mistake was on us for not separating these two things out .... 2 years ago.

Comment (wlach) [reply tests/test_glean_checks.py:28]: Yep, though it's hard to see the implications of doing things the quick way in the beginning. Happy we're in a better place now, at least.

**Pull request merged**